### ISSUE TITLE
Allow small machines to create transactions

### ISSUE THREAD

**Issue by Sword-Smith** [Neptune-Crypto/neptune-core#192]
This logic is expensive to execute:

```rust
```

Adding a path through this code where only the lock script satisfaction is proven will allow weak machines to initiate (non-privacy-preserving) transactions.

**Comment by dan-da** [comment]
I think there should be some discussion around this and what the privacy implications are before we implement.

anonymity comes from hiding in a sea of similar tx.

history:  zcash has strong opt-in privacy.  monero has weaker mandatory privacy.    Zcash users usually didn't opt in, so the anonymity pool was small.   Thus for practical usage monero's privacy was stronger because all tx were equally private and no tx stand out from others, and because the privacy-minded community values this, so the anonymity pool is relatively large.  in the long run, monero's dev community has become so large that now they are implementing full-chain-membership-proofs, ie full utxo-set privacy, equivalent to zcash... except mandatory.

If we enable weak machines to send tx with weak privacy, I imagine those would soon become the dominant tx on the network.  this weakens the anon pool, and weakens neptune's brand as a privacy coin.

I think it is better to insist on strong privacy and do everything we can to enable that.  paying stronger nodes to perform the proofs, using gpus, etc, etc.

so, concept nack from me.

**Comment by aszepieniec** [comment]
I tend to agree with Dan. The option to produce transactions cheaper but in exchange for sacrificing privacy is going to undermine the privacy of users who don't opt out. However, I'm not entirely on one side of this issue, because:
 - Opting out with opt-out feature, and opting out via an alternative cryptocurrency, have largely the same effect on the privacy of users who opt in.
 - An alternative to broadcasting an unprivate transaction is to narrowcast it to a select contract counterparty who is eager to earn a commission for his proving capability. No privacy is lost, assuming an integral counterparty.
 - We can do nothing to prevent this narrowcast strategy from being developed independently and organically growing into a marketplace.
 - Forget about distinct contractual counterparties; different physical machines operated by the same person might want to engage in this overlay protocol.
 - The market is king, and let's listen and respond to what users actually demand. (Which might be different from what we expect even when that expectation is based on reason and evidence.)

In terms of priorities, and with the caveat that market feedback trumps all:
 - We have strategies in the pipeline for making VM faster and require less memory. Let's focus on those first.
 - There are cryptographic solutions for outsourcing work without leaking secrets. FHE is not really practical unless with hardware support. MPC is more efficient but comes at the expense of interactivity. It does not (yet?) achieve work asymmetry, meaning that outsourcer and outsourcee need to do the same amount of work. I bet future research can address these limitations. In fact, I'm sure we can come up with something. Or I can poke a few people who might.
